akka-stream-tests/src/test/scala/akka/stream/scaladsl/AttributesSpec.scala

```diff
@@ -305,6 +260,78 @@ class AttributesSpec extends StreamSpec(ConfigFactory.parseString(
       dispatcher should startWith("AttributesSpec-my-dispatcher")
     }
 
+    "change dispatcher when defined on the async call" in {
+      val dispatcher =
+        Source.fromGraph(
+          new ThreadNameSnitchingStage("akka.stream.default-blocking-io-dispatcher"))
+          .async("my-dispatcher")
+          .runWith(Sink.head)
+          .futureValue
+
+      dispatcher should startWith("AttributesSpec-my-dispatcher")
+    }
+  }
+
+  "attributes on a Flow" must {
+
+    "make the attributes on fromGraph(flow-stage) Flow behave the same as the stage itself" in {
+      val attributes =
+        Source.empty
+          .viaMat(
+            Flow.fromGraph(new AttributesFlow(Attributes.name("original-name")))
+              .withAttributes(Attributes.name("replaced")) // this actually replaces now
```

Comment:
Do we really lose the original name then? That’s too bad hmmm it means we need to keep the add?

Reply:
I think the add is what we keep if we get rid of one it is `withAttributes` as it seldom makes sense to throw away all previously set attributes

Reply:
Yeah i think that could work. <small>esp. if one could override specifics selectively anyway...
</small>